qemu/tests/block_resize.py

```diff
@@ -1,171 +1,148 @@
+                utils_disk.resize_partition_linux(session, partition, str(block_size))
 
         error_context.context("Change disk size to %s in monitor"
                               % block_size, logging.info)
         vm.monitor.block_resize(data_image_dev, block_size)
 
-        if need_reboot:
+        if params.get("guest_prepare_cmd"):
+            session.cmd(params.get("guest_prepare_cmd"))
+        if params.get("need_reboot") == "yes":
             session = vm.reboot(session=session)
-        elif disk_rescan_cmd:
-            error_context.context("Rescan disk", logging.info)
-            session.cmd(disk_rescan_cmd)
+        if params.get("need_rescan") == "yes":
+            drive.rescan_disks(session)
 
-        # We need expand disk in monitor first than extend it in guest
-        if block_size > old_block_size and disk_update_cmd:
+        # We need extend disk in monitor first than extend it in guest
+        if float(ratio) == 1.5:
             error_context.context("Extend disk to %s in guest"
                                   % block_size, logging.info)
-            session.cmd(disk_update_cmd[index])
-
+            if os_type == 'windows':
+                drive.extend_volume(session, mpoint)
+            else:
+                utils_disk.resize_partition_linux(session, partition, str(block_size))
+                utils_disk.resize_filesystem_linux(session, partition,
+                                                   utils_disk.AVAILABLE_SIZE)
         global current_size
         current_size = 0
-        if not utils_misc.wait_for(lambda: compare_block_size
-                                   (session, block_size_cmd,
-                                    block_size_pattern),
-                                   20, 0, 1, "Block Resizing"):
-            test.fail("Block size get from guest is not"
-                      "the same as expected \n"
-                      "Reported: %s\n"
-                      "Expect: %s\n" % (current_size, block_size))
+        if not utils_misc.wait_for(lambda: verify_disk_size(session, os_type,
+                                   disk), 20, 0, 1, "Block Resizing"):
+            test.fail("Block size get from guest is not same as expected.\n"
+                      "Reported: %s\nExpect: %s\n" % (current_size, block_size))
+        session = vm.reboot(session=session)
+
+        if os_type == 'linux':
+            if not utils_disk.is_mount(partition, dst=mpoint,
+                                       fstype=fstype, session=session):
+                utils_disk.mount(partition, mpoint,
+                                 fstype=fstype, session=session)
+
+        if params.get('iozone_test') == 'yes':
+            iozone_timeout = float(params.get("iozone_timeout", 1800))
+            iozone_cmd_options = params.get("iozone_option") % mpoint
+            try:
+                io_test = generate_instance(params, session, 'iozone')
```

Comment:
Hit this issue \"ERROR: local variable \'io_test\' referenced before assignment\", please check the args \"session\"

Reply:
has updated, thanks~